File: DriftGame/uracer-desktop/src/com/bitfire/uracer/game/world/WorldDefs.java
package com.bitfire.uracer.game.world;

public final class WorldDefs {

    /**
     * Attributes for Tiled's object properties.
     */
    public enum ObjectProperties {
        MeshScale("scale");

        public final String mnemonic;

        ObjectProperties(String mnemonic) {
            this.mnemonic = mnemonic;
        }
    }

    /**
     * Tiled's world tile layers
     */
    public enum Layer {

        Track("track"),
        ;


        public final String mnemonic;

        Layer(String mnemonic) {
            this.mnemonic = mnemonic;
        }
    }

    /**
     * Tiled's world object layer/group
     */
    public enum ObjectGroup {
        Lights("lights"),
        StaticMeshes("static-meshes"),
        Trees("trees"),
        Walls("walls"),
        Route("route"),
        Sectors("sectors");

        public final String mnemonic;

        ObjectGroup(String mnemonic) {
            this.mnemonic = mnemonic;
        }
    }
}

File: DriftGame/uracer-desktop/src/com/bitfire/uracer/game/world/WorldDefs.kt
package com.bitfire.uracer.game.world

class WorldDefs {

    /**
     * Attributes for Tiled's object properties.
     */
    enum class ObjectProperties(@JvmField val mnemonic: String) {
        MeshScale("scale");
    }

    /**
     * Tiled's world tile layers
     */
    enum class Layer(@JvmField val mnemonic: String) {
        Track("track");
    }

    /**
     * Tiled's world object layer/group
     */
    enum class ObjectGroup(@JvmField val mnemonic: String) {
        Lights("lights"),
        StaticMeshes("static-meshes"),
        Trees("trees"),
        Walls("walls"),
        Route("route"),
        Sectors("sectors");
    }
}
